python/mpt/tests/src/mpt/test_trie_diff.py

```diff
@@ -139,6 +227,44 @@ def test_trie_diff(
                 keys_in_address += 1
             assert keys_in_address == len(state_diff._storage_tries[address].keys())
 
+    @pytest.mark.parametrize(
+        "invalid_case",
+        [
+            "expected an empty bytes value",
+            "expected at least two non-null subnodes",
+            "expected a non-zero key segment",
+            "expected a non-empty subnode",
+            "expected a non-empty value",
+            "expected a 32-byte combined path and rest of key",
+        ],
+    )
+    @given(data=st.data())
+    def test_invalid_trie_diff(
+        self,
+        cairo_run,
+        empty_ethereum_tries,
+        data,
+        invalid_case,
+    ):
+        invalid_post_trie = data.draw(invalid_mpt_strategy(invalid_case=invalid_case))
+
+        trie = EthereumTrieTransitionDB.from_pre_and_post_tries(
+            empty_ethereum_tries, invalid_post_trie
+        )
+        with pytest.raises(Exception) as cairo_error:
```

Comment:
match the error string

Reply:
the cairo error does not return a string, just ValueError when the check fails

Reply:
I'd rather have us return explicit error strings in this case as we want to make sure we're targetting this exact error. In the future we could interpret the string `ValueError: error_cause` as `ValueError("error_cause")` for these kinds of things

Reply:
how do we proceed?
we replace 
```
    if (bytes_variant.len != 0) {
        raise('ValueError');
    }
```
for instance by
```
with_attr error_message("Non empty bytes value") {
    assert 0 = 1;
}
```
?

Reply:
An idea would be
```
with_attr error_message("ValueError: Non empty bytes value") {
    assert_not_zero(bytes_variant.len)
}
```

Ideally even `raise('ValueError: Non empty bytes value')` but it wont fit in a short string. Or we use error codes.

Reply:
For the time being we could just `raise('NonEmptyBytesValue')`

Reply:
working on a fix